### PR TITLE
Rust analzyer clippy cleanup

### PR DESCRIPTION
**Pull request by jferrant**
I added some allows that should be carefully considered at a later date. (should_implement_trait should not probably be allowed. better solution would be to implement the FromStr trait, but I wanted this to be the minimal amount of changes to prevent Clippy screaming at me)

### CODE REVIEW

**Review by sergey-shandar — APPROVED** [review]

File: frost-signer/src/net.rs [hunk 1]
```diff
@@ -56,17 +56,14 @@ impl NetListen for HttpNetListen {
...
-                        match bincode::deserialize_from::<_, Message>(response.into_reader()) {
-                            Ok(msg) => {
-                                debug!("received {:?}", msg);
-                                self.in_queue.push(msg);
-                            }
-                            Err(_e) => {}
-                        };
-                    }
-                    _ => {}
+                if response.status() == 200 {
```
Comment (donpdonp) [inline frost-signer/src/net.rs:59]: The match is there to make it easy to add other response codes.

File: frost-test/src/lib.rs [hunk 2]
```diff
@@ -8,11 +8,12 @@ mod tests {
...
     use wtfrost::{
-        common::{PolyCommitment, PublicNonce},
+        common::PolyCommitment,
         errors::DkgError,
-        v1::{Party, SignatureAggregator, SignatureShare},
+        v1::{Party, SignatureAggregator},
     };
 
+    #[allow(non_snake_case)]
     fn distribute(
```
Comment (donpdonp) [inline frost-test/src/lib.rs:17]: isnt this already in snake_case?

Comment (sergey-shandar) [reply frost-test/src/lib.rs:17]: It's about `A` parameter

Comment (jferrant) [reply frost-test/src/lib.rs:17]: Yes, unfortunately there are a few instances where we have single capital letter vars which make clippy unhappy. (`A` specifically here, but `T` and `N` below.)

File: frost-signer/src/net.rs (hunk 1 above)
Comment (jferrant) [reply frost-signer/src/net.rs:59]: This is true, but until we do add those response codes, I am more on the side of doing a single if. However, I can change it back and add an allow instead if that is the general consensus :)

File: frost-test/src/lib.rs (hunk 2 above)
Comment (donpdonp) [reply frost-test/src/lib.rs:17]: oh Ive been renaming T and N to `threshold` and `total` in my branch. there is probably a better name for A as well.

File: frost-signer/src/net.rs (hunk 1 above)
Comment (donpdonp) [reply frost-signer/src/net.rs:59]: I see. if the goal is to get clippy to be quiet, thats fine to leave it.

File: frost-test/src/lib.rs (hunk 2 above)
Comment (jferrant) [reply frost-test/src/lib.rs:17]: I can make those changes here or let you merge your branch/remove the allow in it?

**Comment by jferrant** [comment]
I see that we don't run clippy in our pull request CI pipeline. Any opinions on me adding it?

File: frost-test/src/lib.rs (hunk 2 above)
Comment (donpdonp) [reply frost-test/src/lib.rs:17]: i can fix later when I merge.

**Comment by sergey-shandar** [comment]
> I see that we don't run clippy in our pull request CI pipeline. Any opinions on me adding it?

Sure. We can merge this PR and then add clippy to CI: https://github.com/Trust-Machines/core-eng/blob/main/.github/workflows/rust.yml#L49

**Comment by sergey-shandar** [comment]
Here's an example https://github.com/Trust-Machines/frost/blob/main/.github/workflows/ci.yml#L51

File: yarpc/src/read_ex.rs [hunk 3]
```diff
@@ -18,8 +18,7 @@ pub trait ReadEx: Read {
         Ok(buf)
     }
     fn read_exact_vec(&mut self, len: usize) -> Result<Vec<u8>, Error> {
-        let mut buf = Vec::with_capacity(len);
-        buf.resize(len, 0);
+        let mut buf = vec![0; len];
```
Comment (sergey-shandar) [inline yarpc/src/read_ex.rs:21]: I'm trying to avoid macros which can be replaced by functions but I'm too lazy to argue with Clippy.

Comment (jferrant) [reply yarpc/src/read_ex.rs:21]: Oh sorry, I didn't know this was a preference. XD

Comment (sergey-shandar) [reply yarpc/src/read_ex.rs:21]: No, it's my personal preferences. I can live with Clippy default settings.

**Pull request merged**